### PR TITLE
[ENH] testing all forecasters `predict_quantiles`, `predict_interval` in-sample

### PR DESCRIPTION
**Pull request by fkiraly**
Bug https://github.com/sktime/sktime/issues/4468 surfaced that probabilistic forecast methods were not tested in-sample.

This PR adds coverage for probabilistic forecast methods `predict_quantiles`, `predict_interval` used in-sample.

This should also surface #4468 in 0.17.1, and pass on 0.17.0 or post-#4469.

Depends on:
* https://github.com/sktime/sktime/pull/4472 to ensure we test all estimators at the same time
* https://github.com/sktime/sktime/pull/4476 for cleaning up the `test_predict_time_index` test
* fixes in https://github.com/sktime/sktime/pull/4469
* fixes in https://github.com/sktime/sktime/pull/4505
* https://github.com/sktime/sktime/pull/4506

### CODE REVIEW

**Comment by fkiraly** [comment]
@yarnabrina, would appreciate your review since you are working on this (or in close vicinity)

**Comment by fkiraly** [comment]
failures coming from:

* `SquaringResiduals`
* `ForecastX`
* `VECM`

File: sktime/forecasting/tests/test_all_forecasters.py [hunk 1]
```diff
@@ -230,6 +230,12 @@ def test_predict_time_index(
             _assert_correct_pred_time_index(y_pred.index, cutoff, fh=fh_int)
             _assert_correct_columns(y_pred, y_train)
 
+            if estimator_instance.get_tag("capability:pred_int"):
```
Comment (yarnabrina) [inline sktime/forecasting/tests/test_all_forecasters.py:233]: clarification: if this check is specifically made, what can cause `NotImplementedError`?

File: sktime/forecasting/tests/test_all_forecasters.py [hunk 2]
```diff
@@ -230,6 +230,12 @@ def test_predict_time_index(
...
+                y_pred_int = estimator_instance.predict_interval()
+                _assert_correct_pred_time_index(y_pred_int.index, cutoff, fh=fh_int)
+                y_pred_q = estimator_instance.predict_quantiles()
+                _assert_correct_pred_time_index(y_pred_q.index, cutoff, fh=fh_int)
+
```
Comment (yarnabrina) [inline sktime/forecasting/tests/test_all_forecasters.py:238]: Consider addition of `predit_proba`?

Comment (fkiraly) [reply sktime/forecasting/tests/test_all_forecasters.py:238]: hm, yes. I thought the same but hesitated doing this here due to the current `tensorflow` dependency of the method.
Maybe splitting the blocks smartly and try/except would be an option.

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/tests/test_all_forecasters.py:233]: I thought something about in-sample or out-of-sample.

But you are right the try/except is a block with an unclear condition, thinking carefully, I do not know for sure why it is there!

So, let's see what happens when we remove it:
https://github.com/sktime/sktime/pull/4476
If it has a good reason to break in cases on `main`, we should probably replace it with a new tag.

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 2 above)
Comment (fkiraly) [reply sktime/forecasting/tests/test_all_forecasters.py]: the same is actually tested in `test_predict_proba` already.

File: sktime/forecasting/tests/test_all_forecasters.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/tests/test_all_forecasters.py]: I assume we consider this addressed by #4476

**Pull request merged**